app/models/corporate_information_page.rb

```diff
@@ -51,8 +51,16 @@ def search_index
   end
 
   def self.search_only
-    # Ensure only CIPs associated with a live Organisation are indexed in search.
-    super.with_organisation_govuk_status("live")
+    live_govuk_status = super.with_organisation_govuk_status("live")
+
+    accessible_other_govuk_status = super.accessible_documents_policy
+      .with_organisation_govuk_status(%w[joining exempt transitioning])
+
+    accessible_devolved_govuk_status = super.accessible_documents_policy
+      .with_organisation_govuk_status("closed")
+      .where(organisations: { govuk_closed_status: "devolved" })
```

Comment:
I'm having a bit of trouble understanding all the states here - it might be worth splitting this into a couple of commits, e.g. starting with just `live_govuk_status.or(accessible_other_govuk_status)` here and then working your way towards the `accessible_devolved_govuk_status`, which could be its own test case (and commit message).

Reply:
I've split up the commits and hopefully that makes things clearer. What do you think?